source/Halibut/Transport/Proxy/HttpProxyClient.cs

```diff
@@ -287,6 +332,42 @@ void SendConnectionCommand(string host, int port)
             if (_respCode != HttpResponseCodes.OK)
                 HandleProxyCommandError(host, port);
         }
+        
+        async Task SendConnectionCommandAsync(string host, int port, CancellationToken cancellationToken)
+        {
+            var stream = TcpClient.GetStream();
+            var connectCmd = GetConnectCmd(host, port);
+            var request = Encoding.ASCII.GetBytes(connectCmd);
+
+            // send the connect request
+            await stream.WriteAsync(request, 0, request.Length, cancellationToken);
+
+            // wait for the proxy server to respond
+            await WaitForDataAsync(stream, cancellationToken);
+
+            // PROXY SERVER RESPONSE
+            // =======================================================================
+            //HTTP/1.0 200 Connection Established<CR><LF>
+            //[.... other HTTP header lines ending with <CR><LF>..
+            //ignore all of them]
+            //<CR><LF>    // Last Empty Line
+
+            // create an byte response array  
+            var response = new byte[TcpClient.ReceiveBufferSize];
+            var sbuilder = new StringBuilder();
+
+            do
+            {
+                var bytes = await stream.ReadAsync(response, 0, TcpClient.ReceiveBufferSize, cancellationToken);
```

Comment:
If the stream is longer than the receive buffer size, won't this do/while get stuck in a loop as the `offset` value is hardcoded to 0?

Reply:
0 is the offset into the `byte[] response`, the returned bytes is the count of how many bytes were read into `response`. The next read will write to the start of the buffer which is read from in the next line into the string builder.

Reply:
Ah OK, I thought `offset` was the 'start reading from here' point in the stream, but I guess the stream position is incremented automatically. No worries!